website_snippet_mass_mailing_partner/static/src/js/website_snippet_mass_mailing_partner.js

```diff
@@ -0,0 +1,57 @@
+/* © 2016 Jairo Llopis <jairo.llopis@tecnativa.com>
+ * License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). */
+
+"use strict";
+(function ($) {
+    openerp.website.snippet.animationRegistry.subscribe.include({
+        start: function(editable_mode) {
+            var self = this;
+            self._super(editable_mode);
+            if (self.$target.find(".js_subscribe_name:visible")) {
+                openerp.jsonRpc('/website_mass_mailing/is_subscriber', 'call', {
+                    list_id: this.$target.data('list-id'),
+                }).then(function (data) {
+                    self.$target.find('.js_subscribe_name:visible')
+                    .val(data.name)
+                    .attr(
+                        "disabled",
+                        Boolean(data.is_subscriber && data.name.length)
+                    );
+                });
+            }
+        },
+        on_click: function() {
```

Comment:
@hbrunn You are right, but if you review the code, I break it because there's no other way. `_super` would do its stuff inside a namespaced promise that is not returned, so there's absolutely no way to know if it succeeded or not.

OTOH I will remove the name update to avoid that case you say, thanks for the insight.